detect_smb.go

```diff
@@ -0,0 +1,21 @@
+package getter
+
+import (
+	"fmt"
+	"strings"
+)
+
+// SmbDetector implements Detector to detect smb paths with //.
+type SmbDetector struct{}
+
+func (d *SmbDetector) Detect(src, pwd string) (string, bool, error) {
+	if len(src) == 0 {
+		return "", false, nil
+	}
+
+	if strings.HasPrefix(src, "//") {
+		// This is a valid smb path and will be also checked as local file by the SmbGetter
+		return fmt.Sprintf("smb:%s", src), true, nil
+	}
```

Comment:
`//` will be ignored by the FileDetector and will be recognized by the SmbDetector and transformed to a smb scheme URL. This is ok because the SmbGetter uses the FileGetter to check first if there's a possible local mount for the path.